raspberry-pi/0/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib, pkgs, ...}:
+
+{
+  boot = {
+    consoleLogLevel = lib.mkDefault 7;
+    extraTTYs = [ "ttyAMA0" ];
+    kernelPackages = lib.mkDefault pkgs.linuxPackages_rpi;
+    kernelParams = [
+      "dwc_otg.lpm_enable=0"
+      "console=ttyAMA0,115200"
+      "rootwait"
+      "elevator=deadline"
+    ];
+    loader = {
+      grub.enable = lib.mkDefault false;
+      generationsDir.enable = lib.mkDefault false;
+      raspberryPi = {
+        enable = lib.mkDefault true;
+        version = lib.mkDefault 1;
+      };
+    };
+  };
+
+  nix.buildCores = 1;
+
+  nixpkgs.config.platform = lib.systems.platforms.raspberrypi;
+
+  # cpufrequtils doesn't build on ARM
+  powerManagement.enable = lib.mkDefault false;
+
+  services.openssh.enable = lib.mkDefault true;
```

Comment:
I would move the content to `raspberry-pi/default.nix` for all versions and only import the default.nix here for better sharing. It is still good to list every generation explicit however because then people don't need to change anything in case we change something for a particular model.